### PR TITLE
Create snippets for reStructuredText

### LINKED ISSUE #440: Snippets for reStructuredText
I noticed that this project does not include snippets for the reStructuredText markup language. I will create a PR including snippets for it.

### PR DESCRIPTION
**Pull request by unrealapex**
Create snippets for the reStructuredText markup language. A majority of these snippets were copied from the existing markdown snippets and had their relevant formatting terminology modified.

Closes #440

### CODE REVIEW

**Review by OkelleyDevelopment — COMMENTED** [review]
Thanks for the PR, generally these changes look ok to me. Have you confirmed everything is working as expected in a local setup too?

**Comment by unrealapex** [comment]
I have not. Oddly I am unable to load snippets. I think this might be due to me having set up my LSP configuration incorrectly. I will try to debug it.

File: snippets/rst.json [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+        "description": "Add title"
+    },
+    "section": {
+        "prefix": ["section", "-"],
+        "body": ["${0}\n------------\n\n"],
+        "description": "Add section"
+    },
+    "Links": {
+        "prefix": ["l", "link", "hyperlink"],
+        "body": ["${1:text}_\n.. _${1:text} ${2:url}${0}"],
```
Comment (unrealapex) [inline snippets/rst.json:19]: I am unsure if the snippet jumps to the second placeholder. Would you be able to verify this?

**Review by unrealapex — COMMENTED** [review]
I have conformed that every other snippet works!

**Review by OkelleyDevelopment — APPROVED** [review]

**Pull request merged**